torch/distributed/_tensor/random.py

```diff
@@ -142,3 +281,32 @@ def _set_offset(new_offset: int, device_mesh: DeviceMesh) -> None:
                 f"DTensor randomness only supports cuda device type, "
                 f"but got {device_mesh.device_type}"
             )
+
+
+def _get_shard_coord(spec: DTensorSpec) -> List[int]:
+    mesh = spec.mesh
+
+    # get rank coordinate
+    rank_coord = mesh.get_coordinate()
+    if rank_coord is None:
+        raise RuntimeError(
+            "Do not support calculating starting offset for DTensor over sub-mesh"
+        )
+
+    return [rank_coord[mesh_dim] if mesh_dim >= 0 else 0 for mesh_dim in spec.dim_map]
+
+
+def _get_shard_size(spec: DTensorSpec) -> List[int]:
```

Comment:
it looks like this function is only being consumed by `calc_shard_linear_idx`, can we merge these two function?

Reply:
I still want to call it in tests and it may be used when computing local tensor shape for all ranks.

Reply:
If we use `compute_local_offset`, I think we don't really need this API? If this util is being used for other purposes, we should keep it, but if it's only used by one, I think we better not exposing too many methods (even it's private)

Reply:
Right, that sounds great! Let me replace my APIs with the util functions. That enables uneven sharding and I'll also add one more logic I missed: pytorch RNG offset must be a multiple of 4. Adding a round logic should solve it.